crates/revm/src/database.rs

```diff
@@ -112,9 +112,17 @@ impl<DB: StateProvider> DatabaseRef for StateProviderDatabase<DB> {
     /// Retrieves the block hash for a given block number.
     ///
     /// Returns `Ok` with the block hash if found, or the default hash otherwise.
-    /// Note: This operation may potentially be unsafe due to the unwrap operation.
     fn block_hash_ref(&self, number: U256) -> Result<B256, Self::Error> {
-        // Note: this unwrap is potentially unsafe
-        Ok(self.0.block_hash(number.try_into().unwrap())?.unwrap_or_default())
+        // Attempt to convert U256 to u64
+        let block_number = match number.try_into() {
+            Ok(value) => value,
+            Err(_) => return Err(Self::Error::ConversionError(number)),
+        };
```

Comment:
wasn't there another case of try_into?

Reply:
Just found another one in the same file which is fixed now. I searched the rest of the codebase and couldn't find any others.